### PR TITLE
Overwrite signal actions

### PR DESCRIPTION
**Pull request by YuzhouGuo**
Solve the dump issue by calling our own signal handler #95

### CODE REVIEW

File: src/signal/cr_signal.cc [hunk 1]
```diff
@@ -112,8 +112,6 @@ static void InstallFailureSignalHandler() {
 
 void InstallSignalHandler() {
     InstallFailureSignalHandler();
-    google::InstallFailureSignalHandler();
-    google::InstallFailureWriter(WriteToGlog);
```
Comment (chenhao94) [inline src/signal/cr_signal.cc:116]: Why removing it?

File: src/signal/cr_signal.cc [hunk 2]
```diff
@@ -112,8 +112,6 @@ static void InstallFailureSignalHandler() {
 
 void InstallSignalHandler() {
     InstallFailureSignalHandler();
-    google::InstallFailureSignalHandler();
```
Comment (chenhao94) [inline src/signal/cr_signal.cc:115]: 不需要删掉，你只要把我们的后注册覆盖掉前面的就好了，这样以后我们什么地方想恢复成glog的handler只要去掉对应的signal就行

File: src/signal/cr_signal.cc [hunk 3]
```diff
@@ -34,6 +34,11 @@ static void WriteToGlog(const char* data, std::size_t size) {
 static const int kFailureSignals[] = {
     SIGINT,
     SIGTRAP,
+    SIGSEGV,
+    SIGILL,
+    SIGFPE,
```
Comment (xkszltl) [inline src/signal/cr_signal.cc:39]: I don't think we have FPE, even div-by-zero is just nan/inf in C++.

File: src/signal/cr_signal.cc [hunk 4]
```diff
@@ -34,6 +34,11 @@ static void WriteToGlog(const char* data, std::size_t size) {
 static const int kFailureSignals[] = {
     SIGINT,
     SIGTRAP,
+    SIGSEGV,
```
Comment (xkszltl) [inline src/signal/cr_signal.cc:37]: All these signal should be sorted, either by their ID, or by name alphabetically.

Comment (xkszltl) [reply src/signal/cr_signal.cc:38]: resolve的时候最好回一下，即便是简单的fixed/done，这样能明确你的处理方式。

**Review by chenhao94 — APPROVED** [review]
LGTM modulo comments at

- https://github.com/cyfitech/cris-core/issues/95#issuecomment-1312241288

You can merge this PR while keep investigating other solutions.

**Pull request merged**